release-21.2: sql: add COPY to sampled_query logging

otan opened the pull request:
Backport 1/1 commits from #86627.

/cc @cockroachdb/release

---

As requested by product, we want COPY to show up on sampled_query. This
commit adds this, whilst fudging a few things presumably more relevant
to "regular" DDL statements.

Release justification: telemetry only change

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/86717)
<!-- Reviewable:end -->

otan commented:
blathers backport 22.1

otan committed:
sql: add COPY to sampled_query logging

As requested by product, we want COPY to show up on sampled_query. This
commit adds this, whilst fudging a few things presumably more relevant
to "regular" DDL statements.

Release justification: telemetry only change

Release note: None

otan commented:
going to redo this - a lot has changed.
